web-registry/src/pages/EcocreditsByAccount/EcocreditsByAccount.tsx

```diff
@@ -36,16 +30,9 @@ export const EcocreditsByAccount = (): JSX.Element => {
   const theme = useTheme();
   const location = useLocation();
 
-  const graphqlClient =
-    useApolloClient() as ApolloClient<NormalizedCacheObject>;
-
-  const { data: partyByAddr } = useQuery(
-    getPartyByAddrQuery({
-      client: graphqlClient,
-      addr: accountAddress ?? '',
-      enabled: !!accountAddress && !!graphqlClient,
-    }),
-  );
+  const { data: partyByAddr } = usePartyByAddrQuery({
+    variables: { addr: accountAddress ?? '' },
+  });
```

Comment:
While this seems to work, I'm not comfortable relying on retries for ensuring that this call will eventually work. If for some reason we exceed the number of retries and the client still has no csrf token due to network latency this call would still fail.
We could check if we have a CSRF token before doing the request instead to ensure that the call would always work with various different network latencies.

Reply:
Ok, I'll try to have a look to see If I can get the react-query version working otherwise we'll use that.

Reply:
@wgwz I was able to fix the issue with react-query, here's the PR: https://github.com/regen-network/regen-web/pull/1877
I think the problem you ran into was that there were several `partyByAddr`queries and they all need the CSRF safeguard to prevent putting invalid data in the cache.